src/components/coin-icon/FastRequestCoinIcon.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useCallback } from 'react';
+// @ts-ignore
+import { CircularProgress } from 'react-native-circular-progress';
+import FastRequestVendorLogoIcon from './FastRequestVendorLogoIcon';
+import { ThemeContextProps } from '@rainbow-me/theme';
+
+const RequestCoinIconSize = 48;
+
+const RequestCoinIcon = ({
+  dappName,
+  expirationColor,
+  imageUrl,
+  percentElapsed,
+  size = RequestCoinIconSize,
+  theme,
+}: {
+  dappName: string;
+  expirationColor: string;
+  imageUrl: string;
+  percentElapsed: number;
+  size?: number;
+  theme: ThemeContextProps;
+}) => {
+  const { colors } = theme;
+  const renderIcon = useCallback(
+    // react-native-circular-progress expects a single function child.
+    () => (
+      <FastRequestVendorLogoIcon
+        backgroundColor={colors.white}
+        borderRadius={size}
+        dappName={dappName}
+        imageUrl={imageUrl}
+      />
+    ),
+    [colors.white, dappName, imageUrl, size]
+  );
+
+  return (
+    <CircularProgress
+      childrenContainerStyle={{
+        overflow: 'visible',
+      }}
```

Comment:
Not sure if it changes anything, but perhaps it is possible to add this to stylesheet? Does that improve anything?

Reply:
What it improves: code follows code style guides better :D